app/components/App/App.js

```diff
@@ -0,0 +1,207 @@
+    if (this.state.startButtonPressed) {
+      return this.setStartLocation(x, y);
+    }
+
+    if (this.state.endButtonPressed) {
+      return this.setEndLocation(x, y);
+    }
+
+    return this.toggleCellAllowedToBeSteppedOn(x, y);
+  }
+
+  toggleCellAllowedToBeSteppedOn(x, y) {
+    console.log('toggling cell state');
+    const { board, } = this.state;
+    const boardCopy = cloneBoard(board);
+    boardCopy[y][x].isAllowedToBeSteppedOn = !this.state.board[y][x].isAllowedToBeSteppedOn;
+    console.log('board copy:');
+    console.log(boardCopy);
+    this.setState({ board: boardCopy, });
+  }
+
+  toggleStartButtonPressed() {
+    this.setState({
+      startButtonPressed: !this.state.startButtonPressed,
+      endButtonPressed: false,
+    });
+  }
+
+  toggleEndButtonPressed() {
+    this.setState({
+      endButtonPressed: !this.state.endButtonPressed,
+      startButtonPressed: false,
+    });
+  }
+
+  findAndDrawPath() {
+    const { pathStartingPoint, pathEndingPoint, board, } = this.state;
+    const boardCopy = cloneBoard(board);
+    const findPathParameters = {
+      field: boardCopy,
+      startPoint: pathStartingPoint,
+      endPoint: pathEndingPoint,
+    };
+    const path = findPath(findPathParameters);
+    const stepCounter = -1;
+    this.drawPath(path, stepCounter);
+  }
+
+  drawPath(pathRemaining, stepNumber) {
+    if (pathRemaining.length === 0) {
+      return;
+    }
+    this.addStepToPathOnBoard(pathRemaining[0], stepNumber + 1);
+    window.setTimeout(this.drawPath.bind(this, pathRemaining.slice(1), stepNumber + 1), 500);
+  }
+
+  addStepToPathOnBoard(step, stepNum) {
+    const { board, } = this.state;
+    const boardCopy = cloneBoard(board);
+    const [ x, y ] = step;
+    boardCopy[y][x].numberOfStepInPath	= stepNum;
+    this.setState({ board: boardCopy, });
+  }
+  render() {
+    const {
+      handleClickOnBoard,
+      toggleStartButtonPressed,
+      toggleEndButtonPressed,
+      findAndDrawPath,
+    } = this;
+    const handleClicks = {
+      handleClickOnBoard,
+      toggleStartButtonPressed,
+      toggleEndButtonPressed,
+      findAndDrawPath,
+    };
+    const {
+      colorA,
+      colorB,
+      colorC,
+      board,
+      pathStartingPoint,
+      pathEndingPoint,
+      startButtonPressed,
+      endButtonPressed,
+    } = this.state;
+    const gameState = {
+      colorA,
+      colorB,
+      colorC,
+      board,
+      pathStartingPoint,
+      pathEndingPoint,
+      startButtonPressed,
+      endButtonPressed,
+    };
+    const gameProps = {
+      state: gameState,
+      handleClicks,
+    };
+    return (
+      <BrowserRouter>
+        <div className="container">
+          <Switch>
+            <Route
+              exact
+              path="/"
+              render={ () => (
+                <Game { ...gameProps } />
+                ) }
+            />
+          </Switch>
+        </div>
+      </BrowserRouter>
+    );
+  }
+}
+
+App.propTypes = {
```

Comment:
Convert to static getter

Reply:
it's prop types verification. the static getter replaced the `DefaultProps`. Unless I'm missing something..

Reply:
I'm not referring to what it is. Only to syntax. Instead of defining a class and later assigning (static) properties to it, theres syntax which you can use.